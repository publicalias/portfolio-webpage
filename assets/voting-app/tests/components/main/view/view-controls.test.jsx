"use strict";

//local imports

const ViewControls = require("../../../../scripts/components/main/view/view-controls");

const { testWrapper } = require("../../../test-helpers");

//global imports

const { newUser } = require("schemas/master");
const { newPoll } = require("schemas/voting-app");
const { testMock } = require("test-helpers/meta-tests");
const { reactTests } = require("test-helpers/react-tests");

//setup

beforeAll(reactTests.setup);

//view controls

describe("view controls", () => {

  const { testMount, testShallow } = testWrapper(ViewControls);

  it("should match snapshot (default)", () => {

    const { wrapper } = testShallow({}, { poll: newPoll() });

    expect(wrapper).toMatchSnapshot();

  });

  it("should match snapshot (authenticated)", () => {

    const { wrapper } = testShallow({ user: newUser() }, { poll: newPoll() });

    expect(wrapper).toMatchSnapshot();

  });

  it("should call writeText on click", () => {

    const { wrapper } = testMount({}, { poll: newPoll() });

    const writeText = jest.fn();

    navigator.clipboard = { writeText };

    wrapper.find(".qa-share-poll").simulate("click");

    testMock(writeText, [location.href]);

    wrapper.unmount();

  });

});
