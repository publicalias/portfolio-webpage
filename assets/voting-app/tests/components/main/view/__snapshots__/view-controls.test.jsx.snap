// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`view controls should match snapshot (authenticated) 1`] = `
<div
  className="c-view-menu__display-box u-flex-right"
>
  <button
    className="c-view-menu__toggle-btn qa-share-poll"
    onClick={[Function]}
  >
    Share
  </button>
  <div
    className="c-view-menu__toggle-btn"
  >
    <PollToggle
      actions={
        Object {
          "formAddOption": [MockFunction],
          "formClearState": [MockFunction],
          "formRemoveOption": [MockFunction],
          "formSetAdd": [MockFunction],
          "formSetTitle": [MockFunction],
          "formToggleConfirm": [MockFunction],
          "formToggleSecret": [MockFunction],
          "listClearState": [MockFunction],
          "listSetSearch": [MockFunction],
          "metaAddErrors": [MockFunction],
          "metaCloseError": [MockFunction],
          "metaCreatePoll": [MockFunction],
          "metaDeletePoll": [MockFunction],
          "metaGetPolls": [MockFunction],
          "metaGetUser": [MockFunction],
          "metaNoOp": [MockFunction],
          "metaSetLoading": [MockFunction],
          "metaSetState": [MockFunction],
          "metaTimeoutError": [MockFunction],
          "pollAddOption": [MockFunction],
          "pollCastVote": [MockFunction],
          "pollRemoveOption": [MockFunction],
          "pollToggleFlag": [MockFunction],
          "pollToggleHide": [MockFunction],
          "pollToggleSecret": [MockFunction],
          "viewClearState": [MockFunction],
          "viewSetAdd": [MockFunction],
          "viewToggleConfirm": [MockFunction],
          "viewToggleSettings": [MockFunction],
        }
      }
      data={
        Object {
          "errors": Array [],
          "form": Object {
            "add": "",
            "confirm": false,
            "options": Array [],
            "secret": false,
            "title": "",
          },
          "list": Object {
            "search": "",
          },
          "loading": false,
          "polls": Array [],
          "user": Object {
            "auth": Object {
              "id": "",
              "provider": "",
            },
            "data": Object {
              "restricted": false,
            },
            "id": "",
            "name": "",
            "type": "auth",
          },
          "view": Object {
            "add": "",
            "confirm": false,
            "settings": false,
          },
        }
      }
      history={
        Object {
          "push": [MockFunction],
        }
      }
      location={Object {}}
      match={Object {}}
      poll={
        Object {
          "author": "",
          "date": 0,
          "id": "",
          "options": Array [],
          "secret": false,
          "title": "",
          "users": Object {
            "created": "",
            "flagged": Array [],
            "hidden": Array [],
            "voted": 0,
          },
        }
      }
      staticContext={Object {}}
      type="hide"
    />
  </div>
  <div
    className="c-view-menu__toggle-btn"
  >
    <PollToggle
      actions={
        Object {
          "formAddOption": [MockFunction],
          "formClearState": [MockFunction],
          "formRemoveOption": [MockFunction],
          "formSetAdd": [MockFunction],
          "formSetTitle": [MockFunction],
          "formToggleConfirm": [MockFunction],
          "formToggleSecret": [MockFunction],
          "listClearState": [MockFunction],
          "listSetSearch": [MockFunction],
          "metaAddErrors": [MockFunction],
          "metaCloseError": [MockFunction],
          "metaCreatePoll": [MockFunction],
          "metaDeletePoll": [MockFunction],
          "metaGetPolls": [MockFunction],
          "metaGetUser": [MockFunction],
          "metaNoOp": [MockFunction],
          "metaSetLoading": [MockFunction],
          "metaSetState": [MockFunction],
          "metaTimeoutError": [MockFunction],
          "pollAddOption": [MockFunction],
          "pollCastVote": [MockFunction],
          "pollRemoveOption": [MockFunction],
          "pollToggleFlag": [MockFunction],
          "pollToggleHide": [MockFunction],
          "pollToggleSecret": [MockFunction],
          "viewClearState": [MockFunction],
          "viewSetAdd": [MockFunction],
          "viewToggleConfirm": [MockFunction],
          "viewToggleSettings": [MockFunction],
        }
      }
      data={
        Object {
          "errors": Array [],
          "form": Object {
            "add": "",
            "confirm": false,
            "options": Array [],
            "secret": false,
            "title": "",
          },
          "list": Object {
            "search": "",
          },
          "loading": false,
          "polls": Array [],
          "user": Object {
            "auth": Object {
              "id": "",
              "provider": "",
            },
            "data": Object {
              "restricted": false,
            },
            "id": "",
            "name": "",
            "type": "auth",
          },
          "view": Object {
            "add": "",
            "confirm": false,
            "settings": false,
          },
        }
      }
      history={
        Object {
          "push": [MockFunction],
        }
      }
      location={Object {}}
      match={Object {}}
      poll={
        Object {
          "author": "",
          "date": 0,
          "id": "",
          "options": Array [],
          "secret": false,
          "title": "",
          "users": Object {
            "created": "",
            "flagged": Array [],
            "hidden": Array [],
            "voted": 0,
          },
        }
      }
      staticContext={Object {}}
      type="flag"
    />
  </div>
</div>
`;

exports[`view controls should match snapshot (default) 1`] = `
<div
  className="c-view-menu__display-box u-flex-right"
>
  <button
    className="c-view-menu__toggle-btn qa-share-poll"
    onClick={[Function]}
  >
    Share
  </button>
  <div
    className="c-view-menu__toggle-btn"
  >
    <PollToggle
      actions={
        Object {
          "formAddOption": [MockFunction],
          "formClearState": [MockFunction],
          "formRemoveOption": [MockFunction],
          "formSetAdd": [MockFunction],
          "formSetTitle": [MockFunction],
          "formToggleConfirm": [MockFunction],
          "formToggleSecret": [MockFunction],
          "listClearState": [MockFunction],
          "listSetSearch": [MockFunction],
          "metaAddErrors": [MockFunction],
          "metaCloseError": [MockFunction],
          "metaCreatePoll": [MockFunction],
          "metaDeletePoll": [MockFunction],
          "metaGetPolls": [MockFunction],
          "metaGetUser": [MockFunction],
          "metaNoOp": [MockFunction],
          "metaSetLoading": [MockFunction],
          "metaSetState": [MockFunction],
          "metaTimeoutError": [MockFunction],
          "pollAddOption": [MockFunction],
          "pollCastVote": [MockFunction],
          "pollRemoveOption": [MockFunction],
          "pollToggleFlag": [MockFunction],
          "pollToggleHide": [MockFunction],
          "pollToggleSecret": [MockFunction],
          "viewClearState": [MockFunction],
          "viewSetAdd": [MockFunction],
          "viewToggleConfirm": [MockFunction],
          "viewToggleSettings": [MockFunction],
        }
      }
      data={
        Object {
          "errors": Array [],
          "form": Object {
            "add": "",
            "confirm": false,
            "options": Array [],
            "secret": false,
            "title": "",
          },
          "list": Object {
            "search": "",
          },
          "loading": false,
          "polls": Array [],
          "user": Object {},
          "view": Object {
            "add": "",
            "confirm": false,
            "settings": false,
          },
        }
      }
      history={
        Object {
          "push": [MockFunction],
        }
      }
      location={Object {}}
      match={Object {}}
      poll={
        Object {
          "author": "",
          "date": 0,
          "id": "",
          "options": Array [],
          "secret": false,
          "title": "",
          "users": Object {
            "created": "",
            "flagged": Array [],
            "hidden": Array [],
            "voted": 0,
          },
        }
      }
      staticContext={Object {}}
      type="hide"
    />
  </div>
  <div
    className="c-view-menu__toggle-btn"
  />
</div>
`;
